crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -174,6 +175,15 @@ pub struct Res<'w, T: Component> {
     change_tick: u32,
 }
 
+impl<'w, T: Component> Debug for Res<'w, T>
+where
+    T: Debug,
+{
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_fmt(format_args!("Res({:?})", self.value))
```

Comment:
rust debug actually has helper for that https://doc.rust-lang.org/std/fmt/struct.Formatter.html#method.debug_tuple for tuple
```
        fmt.debug_tuple("Res")
            .field(&self.value)
            .finish()
```
or https://doc.rust-lang.org/std/fmt/struct.Formatter.html#method.debug_struct for struct

they let caller control more of the formatting, like precision, pretty print, ...